CMakeLists.txt

```diff
@@ -65,8 +65,14 @@ add_subdirectory(test)
 
 install(PROGRAMS ${CLANG_OCL} DESTINATION bin)
 
-set(CPACK_DEBIAN_PACKAGE_DEPENDS "rocm-llvm, rocm-opencl-dev")
-set(CPACK_RPM_PACKAGE_REQUIRES "rocm-llvm, rocm-opencl-devel")
+set(CPACK_DEBIAN_PACKAGE_DEPENDS "rocm-llvm, rocm-opencl-dev, rocm-core")
+set(CPACK_RPM_PACKAGE_REQUIRES "rocm-llvm, rocm-opencl-devel, rocm-core")
+# Remove dependency on rocm-core if -DROCM_DEP_ROCMCORE=ON not given to cmake
+if(NOT ROCM_DEP_ROCMCORE)
```

Comment:
I think it would be better to do `if(ROCM_DEP_ROCMCORE)` and then do `string(APPEND)` to add rocm-core rather then trying to remove it with regexes.

Reply:
As we discussed briefly in IM, there are special cases, in particular if CPACK_DEBIAN_PACKAGE_DEPENDS is empty then you want to set it to rocm-core, whilst if it is not you want to append a command and then rocm-core. Using regular expressions allows us to do this concisely.

If we knew we were using cmake >= 3.12 we could use
string(JOIN ", " .CPACK_DEBIAN_PACKAGE_DEPENDS  "rocm-core" $CPACK_DEBIAN_PACKAGE_DEPENDS)
to prepend the value correctly. It doesn't seem worthwhile to write our own function to do this, with alternative implementations depending on the cmake version when we have an existing one line solution.